prestoadmin/prestoclient.py

```diff
@@ -191,3 +219,91 @@ def get_rows(self, num_of_rows=NUM_ROWS):
+        result = HTTPSCaCertConnection(
+                self.server, https_port, None, None, ca_file_path, False, URL_TIMEOUT_MS)
+        return result
+
+    def _get_local_keystore_path(self):
+        remote_keystore_path = self.coordinator_config.get_client_keystore_path()
+        keystore_name = os.path.basename(remote_keystore_path)
+        return os.path.join(get_coordinator_directory(), keystore_name)
+
+    def _fetch_keystore(self):
+        remote_keystore_path = self.coordinator_config.get_client_keystore_path()
+        with settings(host_string="%s@%s" % (env.user, env.roledefs['coordinator'][0])):
+            get(remote_keystore_path, self._get_local_keystore_path(), use_sudo=True)
+
+    def _pem_string(self, der_bytes, type):
+        result = "-----BEGIN %s-----\n" % type
+        result += "\r\n".join(
+            textwrap.wrap(base64.b64encode(der_bytes).decode('ascii'), 64))
+        result += "\n-----END %s-----\n" % type
+        return result
+
+    def _write_pem_file(self, directory, der_bytes_list, type):
+        suffix = os.path.join(directory,
+                              '%s-' % type.lower().replace(' ', '-'))
+        fd, pem_path = mkstemp('.pem', suffix)
+        # https://www.digicert.com/ssl-support/pem-ssl-creation.htm
+        with open(pem_path, 'w') as pem_file:
+            for der_bytes in der_bytes_list:
+                pem_file.write(self._pem_string(der_bytes, type))
+        os.close(fd)
+        return pem_path
+
+    def get_pem(self, pa_config=None):
+        local_keystore_path = self._get_local_keystore_path()
+        if not os.path.isfile(local_keystore_path):
+            self._fetch_keystore()
+
+        keystore = jks.KeyStore.load(
+                local_keystore_path,
+                self.coordinator_config.get_client_keystore_password())
+
+        if len(keystore.private_keys.items()) == 0:
+            _, private_key = keystore.private_keys.items()[0]
+        else:
+            private_key = self._get_private_key(keystore, pa_config)
+
+        if not self.ca_file_path:
+            self.ca_file_path = self._write_pem_file(
+                    get_coordinator_directory(),
+                    [cert[1] for cert in private_key.cert_chain], 'CERTIFICATE')
+
+        return self.ca_file_path
+
+    def _get_private_key(self, keystore, pa_config=None):
+        try:
+            alias = pa_config[CERTIFICATE_ALIAS]
+            return keystore.private_keys[alias]
+        except KeyError:
+            alias = prompt_choices(keystore.private_keys.keys())
+            pa_config[CERTIFICATE_ALIAS] = alias
+            return keystore.private_keys[alias]
+
+
+class AliasConfigItem(object):
+    def __init__(self):
+        self.key = CERTIFICATE_ALIAS
+
+    def prompt_user(self, conf):
+        with closing(PrestoClient(None, None)) as client:
+            if not client.requires_https():
+                return
+            client.get_pem(conf)
```

Comment:
Maybe move the KeyError catch here to avoid the runtime prompt error, which is misleading?